src/controllers/user/handlers.js

```diff
@@ -1,18 +1,35 @@
 'use strict'
 
-const { encryptString } = require('../../_utils/hashing')
-const { save, read, delOne } = require('./queries.js')
+const ObjectId = require('mongoose').Types.ObjectId
+
+const { create, read, deleteOne, updateOne } = require('./queries.js')
 const { USER_ACCOUNT_TYPE } = require('../../_utils/constants')
+const { encryptString } = require('../../_utils/hashing')
+const { generateUniqueToken } = require('../../_utils/helpers')
 
 module.exports = {
-  createUser: async (data) => {
-    const { email, firstName, lastName, password } = data
-
+  /**
+   * @param {string} email user email
+   * @param {string} firstName user first name
+   * @param {string} lastName user last name
+   * @param {string} phone user phone
+   * @param {string} password user password
+   * @param {boolean} isCoach If the user is a coach
```

Comment:
FLow ot be defined, but the way I see it and the way it is at the moment, the onboarding flow is the same for coaches and other users. Once logged in, you can "become a coach", then we would set the flag to true. A bit like airbnb. I think it makes things easier and is more user friendly. to be discussed.

Reply:
@alvermeylen You're right, I'm for having different onboarding depend of each cases